pystac_client/item_search.py

```diff
@@ -329,6 +329,8 @@ def _clean_params_for_get_request(self) -> Dict[str, Any]:
             params["collections"] = ",".join(params["collections"])
         if "intersects" in params:
             params["intersects"] = json.dumps(params["intersects"])
```

Comment:
intersects should also use the same `separators` parameters for the compact represetnation

Reply:
Implemented here: https://github.com/stac-utils/pystac-client/pull/337/files#diff-fe17dddc82d77792572abf5f8f958d74c5a701313fd1f842c9b66364a77aae51R331-R333